src/thunderbots/software/ai/hl/stp/evaluation/indirect_chip.h

```diff
@@ -0,0 +1,144 @@
+#include "ai/world/world.h"
+#include "geom/angle.h"
+#include "geom/point.h"
+#include "util/parameter/dynamic_parameters.h"
+
+/**
+ * The indirect chip and chase target evaluation function returns a target point for the
+ * chipper and chaser.
+ * Other related evaluation functions are also included in this file.
+ */
+namespace Evaluation
+{
+    typedef Point Vector2;
+    template <size_t N>
+    using Poly     = std::array<Vector2, N>;
+    using Triangle = Poly<3>;
+
+    /**
+     * Returns the target point that the chipper and chaser will chip and chase at.
+     *
+     * Given the enemy robots' positions, filters and creates a vector of
+     * triangles to be used as a parameter. The target point is where ball will land
+     * according to chipping calibration, as well as where the chaser will meet the ball
+     * at.
+     *
+     * @param world The world in which we want to find the target point
+     *
+     * @return Point to chip and chase at; If null, the target triangles were empty
```

Comment:
Do we mean the triangles were empty as in there were no enemies in them? Or just that we could not find an open triangle to chip to?

Reply:
Ah, I didn't catch this comment earlier, I am actually wondering the same thing:
> The target point would be "null" if the target triangles were empty. However, I'm actually confused too about what "empty" means. The target triangles are previously filtered before attempting to find a target point, so that they are open triangles that do not consist of enemy robots. Wouldn't that mean it is correct for the triangles to be empty?

The `if` statement checks if `!(triangles.empty())`. If false, returns "null". I assume that this is checking if the triangle is empty. If so, I thought it would make more sense the other way around (the `if` statement could serve as a double check before actually going ahead to find the target point within the triangle).

Taken from the earlier comment [here](https://github.com/UBC-Thunderbots/Software/pull/351#discussion_r265418138).

Reply:
Just gonna link to my response to another comment since I think I ended up resolving this there
https://github.com/UBC-Thunderbots/Software/pull/351#discussion_r265573155